runtime/src/main/cpp/ObjCExport.mm

```diff
@@ -417,7 +418,11 @@ -(ObjHeader*)toKotlin:(ObjHeader**)OBJ_RESULT {
     case 'i': RETURN_RESULT_OF(Kotlin_boxInt, self.intValue);
     case 'q': RETURN_RESULT_OF(Kotlin_boxLong, self.longLongValue);
     case 'f': RETURN_RESULT_OF(Kotlin_boxFloat, self.floatValue);
-    case 'd': RETURN_RESULT_OF(Kotlin_boxDouble, self.doubleValue);
+    case 'd': if ([self isKindOfClass:[NSDecimalNumber class]]) {
```

Comment:
Is this conversion intended to be fast?

Reply:
Yes, and I've just realized that it's possible to add `toKotlin:` method to `NSDecimalNumber` class instead.